### PR TITLE
369687 : Custom Role to read cluster

### PR DESCRIPTION
**Pull request by rajeshkaremane**
# **What this PR does / why we need it**:
*A brief description of changes being made*
*Link to the relevant work items: *
[AB#369687](https://dev.azure.com/defragovuk/93c65135-d16b-49d6-a191-4a5313532779/_workitems/edit/369687)

depend on 
https://github.com/DEFRA/adp-pipeline-common/pull/179

# **Special notes for your reviewer**
*Any specific actions or notes on review?*

# Testing
*Any relevant testing information and pipeline runs.*
https://dev.azure.com/defragovuk/DEFRA-FFC/_build/results?buildId=541531&view=results
image

# Checklist (please delete before completing or setting auto-complete)
- [x] Story Work items associated (not Tasks)
- [x] Successful testing run(s) link provided
- [x] Title pattern should be `{work item number}: {title}`
- [x] Description covers all the changes in the PR
- [ ] This PR contains documentation
- [ ] This PR contains tests

# Dependency check (tick or delete)
Have you updated any related templates, repositories or dependencies for new changes? We must ensure any dependencies like Software Templates are updated to include the latest changes to be scaffolded for all services where applicable.

- [ ] ADP Software Templates? - https://github.com/DEFRA/adp-software-templates
- [ ] ADP Portal Web API OR UI? https://github.com/DEFRA/adp-portal-api https://github.com/DEFRA/adp-portal 

# **How does this PR make you feel**:
gif

### CODE REVIEW

**Comment by DaRoszko** [comment]
@rajeshkaremane  have we checked this role worked / and is sufficient, when combined with the Cluster RBAC Reader wrong? (not writer or Admin)

**Review by DaRoszko — COMMENTED** [review]
@rajeshkaremane have we checked this role worked / and is sufficient, when combined with the Cluster RBAC Reader wrong? (not writer or Admin)

**Review by DaRoszko — APPROVED** [review]

File: infra/core/tenant/role-definition/cluster-custom-resources-reader-role.bicep [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+@description('Name of the Custom Role Definition in Azure.')
+param roleName string
+
+@description('Array of scopes to assign the role to.')
+param roleScopes array
+
+@description('Detailed description of the role definition')
+param roleDescription string = 'Custom Role to read cluster data.'
+
+resource asbCustomRoleResource 'Microsoft.Authorization/roleDefinitions@2022-04-01' = {
```
Comment (ajaysoni-cap) [inline infra/core/tenant/role-definition/cluster-custom-resources-reader-role.bicep:12]: why is the resource named as asbCustomRoleResource? could you correct the name please.

Comment (rajeshkaremane) [reply infra/core/tenant/role-definition/cluster-custom-resources-reader-role.bicep]: Fixed

**Review by ajaysoni-cap — APPROVED** [review]

**Review by DaRoszko — APPROVED** [review]

**Pull request merged**